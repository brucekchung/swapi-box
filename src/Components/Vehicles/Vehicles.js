import React from 'react'
import Card from '../Card/Card'

const Vehicles = ({ vehicleData, favorite }) => {
  const cards = vehicleData.map((card, index) => 
    ( <Card key={`vehicle${index}`} 
            data={card} 
            type="vehicleCard" 
            favorite={favorite}/> ))

  return (
    <div className="Vehicles">
      { cards }
    </div>
  ) 
}

export default Vehicles