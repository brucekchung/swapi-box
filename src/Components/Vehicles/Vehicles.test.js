import React from 'react'
import { shallow } from 'enzyme'
import Vehicles from './Vehicles'

describe('Vehicles', () => {
  const props = {
    vehicleData: [{
      "model": "Digger Crawler", 
      "name": "Sand Crawler",
    }],
    favorite: () => {},
  }

  it('should match the snapshot', () => {
    const wrapper = shallow(<Vehicles {...props}/>)
    
    expect(wrapper).toMatchSnapshot()
  })
})