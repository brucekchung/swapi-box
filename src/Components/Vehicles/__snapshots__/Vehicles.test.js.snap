// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    favorite={[Function]}
    vehicleData={
        Array [
            Object {
              "model": "Digger Crawler",
              "name": "Sand Crawler",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          data={
                    Object {
                              "model": "Digger Crawler",
                              "name": "Sand Crawler",
                            }
          }
          favorite={[Function]}
          type="vehicleCard"
/>,
      ],
      "className": "Vehicles",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "vehicle0",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "model": "Digger Crawler",
            "name": "Sand Crawler",
          },
          "favorite": [Function],
          "type": "vehicleCard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            data={
                        Object {
                                    "model": "Digger Crawler",
                                    "name": "Sand Crawler",
                                  }
            }
            favorite={[Function]}
            type="vehicleCard"
/>,
        ],
        "className": "Vehicles",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "vehicle0",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "model": "Digger Crawler",
              "name": "Sand Crawler",
            },
            "favorite": [Function],
            "type": "vehicleCard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
